File: javascript 90 questions/Q30.js

let integer=123456;
let numstr=integer.toString();
let length=numstr.length;
let even=0;
let odd=0;
for(i=0;i<length;i++){
    let digit=parseInt(numstr[i]);
    
    if(numstr[i]%2==0){
        even=even+digit;
    }
    else{
        odd=odd+digit;
    }
}
console.log(even);
console.log(odd);

File: C questions/Q30.c


#include <stdio.h>

int main() {
    int n;       
    int evenSum = 0;   
    int oddSum = 0;     
    printf("Enter an integer: ");
    scanf("%d", &n);
    while (n != 0) {
        int digit = n % 10;

        if (digit % 2 == 0) {
            evenSum += digit;
        } else {
            oddSum += digit; 
        }
        n /= 10;
    }
    printf("Sum of even digits: %d\n", evenSum);
    printf("Sum of odd digits: %d\n", oddSum);

    return 0;
}
